apis/v1alpha2/httproute_types.go

```diff
@@ -598,6 +614,24 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// AutoHTTPS if set to true redirects any incoming HTTP requests to HTTPS.
+	// The path and querystring of the request MUST be preserved in the Location
+	// header in the response.
+	AutoHTTPS bool `json:"auto_https,omitempty"`
+	// TODO if defaulting is possible here (not sure about defaulting in filter), default this?
+	// What status codes should be allowed? 301,302,307,308?
+	// TODO check conformance on this field
+	// +kubebuilder:default=302
+	RedirectStatusCode *int `json:"redirect_status_code,omitempty"`
+	// Location denotes the value of location header.
+	// +optional
+	Location *string `json:"location,omitempty"`
```

Comment:
I am a bit concerned with having the raw header here, rather than something a bit more abstract, like
```
    redirect:
      uri: /v1/bookRatings
      authority: newratings.default.svc.cluster.local
```

The reason is that I can no longer make it generic - I cannot just say redirect /v1 to /v2, I need to now also include http vs https, so now the route is aware of the TLS layer which is leaking from Gateway

Reply:
Does it makes sense to have arbitrary headers here (and do proxies support that)? If so, you can use the same name list structure as the other parts of the API.

Reply:
>  I need to now also include http vs https

IME this is not an issue and users want to be as explicit as possible. But you do make a good point that the user now has to be aware of `authority` as well, which could be abstracted away in Gateway for some use cases. I'll update.